client/state/login/actions.js

```diff
@@ -145,21 +160,21 @@ export const loginUserWithTwoFactorVerificationCode = ( two_step_code, remember_
 		.then( () => {
 			dispatch( { type: TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST_SUCCESS } );
 		} )
-		.catch( ( error ) => {
-			const errorMessage = getMessageFromHTTPError( error );
+		.catch( ( httpError ) => {
+			const error = getErrorFromHTTPError( httpError );
 
 			dispatch( {
 				type: TWO_FACTOR_AUTHENTICATION_UPDATE_NONCE,
-				twoStepNonce: get( error, 'response.body.data.two_step_nonce' ),
-				twoFactorAuthType,
+				twoStepNonce: get( httpError, 'response.body.data.two_step_nonce' ),
+				nonceType: twoFactorAuthType,
 			} );
 
 			dispatch( {
 				type: TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST_FAILURE,
-				error: errorMessage,
+				error,
 			} );
 
-			return Promise.reject( errorMessage );
+			return Promise.reject( error );
```

Comment:
You linked to the `loginUserWithTwoFactorVerificationCode()` function but I think you were really talking about the same line in the `loginSocialUser()` function.

I must admit this is just a mistake, and this might have broken the changes introduced by @Tug in https://github.com/Automattic/wp-calypso/pull/14679. I'll create a card to our Trello board to fix it.

Good job noticing it!